drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/api/TransactionManagerHelper.java

```diff
@@ -58,11 +59,15 @@ public static void removeFromUpdatableSet(TransactionManager txm, Transformable
 
     @SuppressWarnings("unchecked")
     public static Set<Transformable> getUpdateableSet(TransactionManager txm) {
-        Set<Transformable> toBeUpdated = (Set<Transformable>) txm.getResource(APP_UPDETEABLE_RESOURCE);
-        if (toBeUpdated == null) {
-            return Collections.emptySet();
+        Set<Transformable> result = (Set<Transformable>) txm.getResource(APP_UPDETEABLE_RESOURCE);
+        if (result != null) {
+            SortedSet<Transformable> sorted = new TreeSet<>((o1, o2) -> {
+                int compared = o1.getClass().getSimpleName().compareTo(o2.getClass().getSimpleName());
+                return compared == 0 ? 1 : compared;
+            });
```

Comment:
Could we simplify the custom Comparator with this?
```suggestion
            SortedSet<Transformable> sorted = new TreeSet<>(Comparator.comparing(o -> o.getClass().getSimpleName()));
```